### PR TITLE
Day 2

### PR DESCRIPTION
**Pull request by Alzubair98**
**All requirements are done** 

- Splash screen
- 
- A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages
- 
- The user should be able to register in the app with full name, email and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (catagories page)
- 
- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page
- 
- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page
- 
- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new transaction" page
- 
- The user fills out a form to create a new transaction with:
- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

**Comment by prolajumokeoni** [comment]
Hi @Alzubair98, Kindly make sure that there is no custom change to your [linters.yml](https://github.com/Alzubair98/budget_app/blob/292fca7f80b85ed37d1b39a7c48e8ad717110ff6/.github/workflows/linters.yml#L23) config file, please uncomment the Stylelint config 👍🏿

File: README.md [hunk 1]
```diff
@@ -1,25 +1,84 @@
...
-* System dependencies
+- Major languages: Ruby
+- Frameworks: Ruby on Rails
+- Technologies used: Rubocop, gem, Postgresql
 
-* Configuration
+## Live Demo (if available)
 
-* Database creation
+[Live Demo Link](https://livedemo.com)
```
Comment (prolajumokeoni) [inline README.md]: - [x] Kindly make sure your project is deployed and accessible online, you can deploy using Heroku, Netlify or GitHub Pages, ensure that the home and detail page works as expected

File: README.md [hunk 2]
```diff
@@ -1,25 +1,84 @@
...
+
+## Acknowledgments
+
+- Hat tip to anyone whose code was used
+- Inspiration
+- etc
+
+## video
+
+- check this (link)[https://drive.google.com/drive/folders/1e5z4YFoYFhpK_oCVQ0g1wnITClikUGVA]
```
Comment (prolajumokeoni) [inline README.md]: - [x] Kindly make sure that you grant access to your video presentation so the next reviewer can review it
Screenshot_531

File: README.md [hunk 3]
```diff
@@ -1,25 +1,84 @@
...
+
+## 📝 License
+
+The design was inspired by [This](https://creativecommons.org/licenses/by-nc/4.0/)
```
Comment (prolajumokeoni) [inline README.md]: - [x] Your  README should contain an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/), currently, this link does not redirect to the design link on Behance

File: app/views/groups/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      <img src="<%= group.icon%>" class='img'>
+    </div>
+    <div class='cont-text'>
+      <h5><%=link_to group.name.capitalize , group_path(group), class:'underline text-color'%></h5>
+      <p>Total Amount :- <%= group.entities.sum(:amount) %> $</p>
+      <%= link_to 'View' , group_path(group), class:'bttn underline text-color'%>
+      <p class="text-color"> created <%= time_ago_in_words(group.created_at) %> ago</p>
+    </div>
+  </div>
+  <%end%>
```
Comment (prolajumokeoni) [inline app/views/groups/index.html.erb:17]: - [x] Kindly make sure that when the user logs in, they are presented with the categories page, currently, they are redirected to the signout page, you can add the signout link as a link in your navigation bar.

File: app/views/groups/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<h1 class="text-center mt-5">Add New Catagory</h1>
```
Comment (prolajumokeoni) [inline app/views/groups/new.html.erb:2]: - [x] Based on the requirements,   please make sure when the user clicks on the "Back" button (<), the user navigates to the home page, there is currently no back button on your page, please add it 👍🏿

File: app/views/groups/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+            <h5 class="card-title text-color"><%= entity.name.capitalize%></h5>
+          </div>
+          <div class="card-text text-color">Amount: <%= entity.amount%>$</div>
+        </div>
+        <div class="card-footer text-muted">
+          <p> created <%= time_ago_in_words(entity.created_at) %> ago</p>
+        </div>
+      </div>
+    </div>
+  </div>
```
Comment (prolajumokeoni) [inline app/views/groups/show.html.erb:19]: - [x] Based on the requirements, please make sure when the user clicks on the "Back" button (<), the user navigates to the home page, there is currently no back button on your page, please add it 👍🏿

- [x] Based on the requirements, when the user click (or taps) the "Save" button to create the new category, and it's taken to the transactions page for that category
- I currently can not save a transaction, see video below ⬇️ 

https://user-images.githubusercontent.com/69638013/202265981-c31589e5-22c6-4c96-9dcd-8afd43cd162b.mp4

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @Alzubair98 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Git flow was correctly used
✔️  Good design 



## Required Changes ♻️


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (Alzubair98) [reply README.md]: okay sorry for that

File: README.md (hunk 1 above)
Comment (Alzubair98) [reply README.md]: done

File: app/views/groups/index.html.erb (hunk 4 above)
Comment (Alzubair98) [reply app/views/groups/index.html.erb:17]: done

File: app/views/groups/new.html.erb (hunk 5 above)
Comment (Alzubair98) [reply app/views/groups/new.html.erb:2]: done

File: app/views/groups/show.html.erb (hunk 6 above)
Comment (Alzubair98) [reply app/views/groups/show.html.erb:19]: done

**Comment by Alzubair98** [comment]
note for the new reviewer please make sure when you create new transaction 
ENTER THE ID OF THE GROUP (OTHERWISE TRANSACTION WILL NOT BE SAVED) 
THANKS 

ALSO HERE IS THE LINKE TO WATCH THE VIDEO 
https://drive.google.com/drive/folders/1e5z4YFoYFhpK_oCVQ0g1wnITClikUGVA?usp=sharing

File: README.md [hunk 7]
```diff
@@ -1,25 +1,82 @@
...
+
+## Acknowledgments
+
+- Hat tip to anyone whose code was used
+- Inspiration by (designer)[https://www.behance.net/gregoirevella]
+- etc
+
+## video
+
+- check this (link)[https://drive.google.com/drive/folders/1e5z4YFoYFhpK_oCVQ0g1wnITClikUGVA?usp=sharing]
```
Comment (tresorsawasawa) [inline README.md:76]: Good job giving access to your presentation video as mentioned by the previous reviewer. So, I noticed that it does match the required features as mentioned in the project's requirements and this should not be marked by the previous reviewer as you didn't give access. Make sure your video matches the below features:
  - The maximum length of the video is 5 minutes.
  - It should contain a highlight about a piece of the code you find interesting or you are proud of making.

File: README.md [hunk 8]
```diff
@@ -1,25 +1,82 @@
...
+Feel free to check the [issues page](../../issues/).
+
+## Show your support
+
+Give a star ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Hat tip to anyone whose code was used
+- Inspiration by (designer)[https://www.behance.net/gregoirevella]
```
Comment (tresorsawasawa) [inline README.md]: - [ **OPTIONAL** ] **IMPORTANT:** Good job adding this change required by the previous reviewer. But to make it sounds better I suggest you write it like this:
```
- Great thanks to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=).
```

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUEST CHANGES :recycle::recycle: 

Hi @Alzubair98 
Great work on making the changes requested by the previous reviewer  👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 8 above)
Comment (Alzubair98) [reply README.md]: thanks dude i will edit it

File: README.md (hunk 7 above)
Comment (Alzubair98) [reply README.md:76]: done

**Comment by Alzubair98** [comment]
VIDEO FIXED 
YOU CAN CHECK THE NEW VIDEO FROM THIS LINK 
https://drive.google.com/drive/folders/1e5z4YFoYFhpK_oCVQ0g1wnITClikUGVA?usp=sharing

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Alzubair98 

Your project is complete! Great job for making the previous reviewers requested changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Requestv✔️ 
- Great design✔️ 
- app functions as expected✔️ 
- Test ✔️ 
- No linter errors✔️ 

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

- [ optional] please always try to give a clear purpose on how your app/project can be of help to the user during presentation. nice work

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**